config/application.rb

```diff
@@ -205,6 +205,10 @@ class Application < Rails::Application
     config.middleware.use Rack::Attack
     config.middleware.use Mastodon::RackMiddleware
 
+    initializer :deprecator do |app|
+      app.deprecators[:mastodon] = ActiveSupport::Deprecation.new('4.3', 'mastodon/mastodon')
```

Comment:
Rails 7.1 changes how managing deprecations work. Instead of having one central access point which all deprecations pass through, apps/gems can register individual deprecators as part of a larger collection. https://edgeguides.rubyonrails.org/7_1_release_notes.html#add-rails-application-deprecators https://github.com/rails/rails/pull/46049 - but its still possible to centrally configure how you want the deprecators to behave (raise, log, etc).

I'm borrowing this specific syntax approach from other examples I've seen of the same upgrade, but if someone had a very strong opinion on naming it differently I'd be open to that.

Reply:
Ok, I was wondering about the `4.3`. Should this be hardcoded? What does it represent?

Reply:
Ok, should we maybe use `app.deprecators[:mastodon_43]` right now so it is explicitly tied to this version?

Reply:
> Ok, should we maybe use `app.deprecators[:mastodon_43]` right now so it is explicitly tied to this version?

I think it mostly doesn't matter, but doing something like that would be slightly more precise, yes.

> Is there a specific benefit of having multiple deprecators for the app (for each version), except being able to quickly find what we marked for deprecation for this version and remove the related code?

I think the bigger benefit is in how you interact with deprecations from gems you depend on ... but yes, there'd be a hypothetical benefit to being able to more quickly locate all the promised deprecations and pull out the warning and turn them into errors or just remove the features or whatever.